src/azul/drs.py

```diff
@@ -74,10 +80,62 @@
     headers: Optional[Mapping[str, str]] = None
 
 
+class IdentifierClient:
+
+    @classmethod
+    @cache
+    def instance(cls) -> 'IdentifierClient':
+        return cls()
```

Comment:
`@cache` It uses a lock, a dictionary, a list, several closures and other objects. It is complicated. The lock limits concurrency. 

I only see an instance of this paired with each instance of DRSClient so if that's already cached, or has low cardinality, there is no need to globally cache instances of this. Otherwise, there should be a motivating comment. I can explain more what our caching practices are in PL if you'd like.